contracts/marketplace/src/execute.rs

```diff
@@ -367,7 +371,14 @@ pub fn execute_set_bid(
             },
         )?;
     } else {
-        // Bid meets ask criteria so finalize sale
+        // Bid meets ask criteria, so check if token is reserved
+        if let Some(reserved_for) = ask.reserve_for {
+            if reserved_for != bidder {
+                return Err(ContractError::TokenReserved {});
+            }
+        }
```

Comment:
can we move this to the top? we could return earlier because I don't think it would make sense to let someone place a bid if it is already reserved?

Reply:
Moved it up higher and did a bit of refactoring.